utils/python/tests/scripts_regression_tests

```diff
@@ -916,6 +916,95 @@ class TestSingleSubmit(TestCreateTestCommon):
                          msg="COMMAND SHOULD HAVE WORKED\nwait_for_tests output:\n%s\n\nerrput:\n%s\n\ncode: %d" % (output, errput, stat))
 
 ###############################################################################
+class XMLQUERY(unittest.TestCase):
+# Testing command line scripts    
+###############################################################################
+    
+    def setUp(self):
+        # Create case directory
+        self._testroot = MACHINE.get_value("CESMSCRATCHROOT")  # "/tmp/" 
+        self._testdirs = []
+        self._do_teardown = []
+        
+        
+        testdir = os.path.join(self._testroot, 'scripts_regression_tests.testscripts')
+        machine = 'edison'
+        if os.path.exists(testdir):
+            shutil.rmtree(testdir)
+        self._testdirs.append(testdir)
+        cmd = "%s/create_newcase --case %s --compset X --res f19_g16 --mach %s " % (SCRIPT_DIR, testdir, machine)
+        stat, output, errput = run_cmd(cmd, ok_to_fail=True, from_dir=SCRIPT_DIR)
+        self.assertEqual(stat, 0, msg="COMMAND '%s' SHOULD HAVE WORKED\noutput:\n%s\n\nerrput:\n%s" % (cmd, output, errput))
+
+        self._do_teardown.append(testdir)
+        
+        
+    def tearDown(self):
+        pass
+        
+        
+    def test_XMLQUERY(self):
+     
+        # Set script and script path    
+	TOOLS_DIR = SCRIPT_DIR + "/Tools"        
+	xmlquery = "./xmlquery"
+        testdir  = self._testdirs[0]
+        
+        
+        # Check for environment 
+        self.assertTrue(os.path.isdir(SCRIPT_DIR))
+        self.assertTrue(os.path.isdir(TOOLS_DIR))
+        self.assertTrue(os.path.isfile(testdir + "/" + xmlquery) )
+        
+        # Test command line options
+        
+        options = [ 
+            [] , 
+            ['--listall'] , 
+            ['--listall' , '--valonly'] , 
+            ['-caseroot' , testdir , '-valonly' ,
+             '-subgroup' , 'case.run' , '-value' ,  'JOB_QUEUE' ] ,
+        ]
+        
+        for opt in options :
+            
+            out = ''
+            call = [xmlquery] + opt
+            out = subprocess.check_output(call , cwd=testdir )
+        
+            self.assertTrue(len(out) , msg="output exists for:\n " + xmlquery + " " + " ".join(opt) + '\n' + out + 'END' )
+    
+    def test_subgroup(self):
+        
+        # Set script and script path    
+        TOOLS_DIR = SCRIPT_DIR + "/Tools"
+        xmlquery = TOOLS_DIR + "/xmlquery"
+        testdir  = self._testdirs[0]
+    
+        options = [
+            ['-caseroot' , testdir , '-valonly' ,
+             '-value' ,  'JOB_QUEUE' ] ,
+            ['-caseroot' , testdir , '-valonly' ,
+             '-subgroup' , 'case.run' , '-value' ,  'JOB_QUEUE' ] ,
+         ]
+        
+        # Get value and group information
+        base_out = ''
+        call     = [xmlquery] + options[0]
+        base_out = subprocess.check_output(call)
```

Comment:
Use run_cmd. Run_cmd runs things in shell mode, so you don't need this awkward handling of options.